extensions/2.0/Khronos/KHR_interactivity/Specification.adoc

```diff
@@ -0,0 +1,2914 @@
+.. Remove this activation index value from the node's and the graph's arrays of activation indices.
+.. Cancel the corresponding scheduled activation.
+
+===== Cancel Delay
+
+[cols="1h,1,2"]
+|===
+| Type | `flow/cancelDelay` | Cancel a previously scheduled output flow activation
+| Input flow sockets
+| `in` | The entry flow into this node
+| Input value sockets
+| `int delayIndex` | The index value of the scheduled activation to be cancelled
+| Output flow sockets
+| `out` | The flow to be activated after executing this node
+|===
+
+This node has no internal state but its execution **MAY** affect internal states of other nodes and the graph.
+
+When the `in` input flow is activated:
+
+1. Evaluate the `delayIndex` input value.
+2. Remove this activation index value from all arrays of activation indices if it exists.
+3. Cancel the corresponding scheduled activation if it exists.
+4. Activate the `out` output flow.
+
+Non-positive or not existing delay index values **MUST NOT** cause any runtime errors.
+
+=== State Manipulation Nodes
+
+==== Custom Variable Access
+
+===== Variable Get
+
+[cols="1h,1,2"]
+|===
+| Type | `variable/get` | Get a custom variable value
+| Configuration
+| `int variable` | The custom variable index
+| Output value sockets
+| `T value`  | The custom variable value
+|===
+
+This node gets a custom variable value using the variable index provided by the `variable` configuration value. The type `T` is determined by the referenced variable. The variable index **MUST** be non-negative and less than the total number of custom variables, otherwise the node is invalid.
+
+This node has no internal state.
+
+===== Variable Set
+
+[cols="1h,1,2"]
+|===
+| Type | `variable/set` | Set a custom variable value
+| Configuration
+| `int variable` | The custom variable index
+| Input flow sockets
+| `in` | The entry flow into this node
+| Input value sockets
+| `T value` | The new variable value
+| Output flow sockets
+| `out` | The flow to be activated after the value is set
+|===
+
+This node sets a custom variable value using the variable index provided by the `variable` configuration value and the `value` input value. The type `T` is determined by the referenced variable. The variable index **MUST** be non-negative and less than the total number of custom variables, otherwise the node is invalid.
+
+This node has no internal state.
+
+When the `in` input flow is activated:
+
+1. Evaluate the `value` input value.
+2. Set the custom variable with the `variable` configuration value index to the `value` input value.
+3. Activate the `out` output flow.
+
+==== Object Model Access
+
+===== Pointer Get
+
+[cols="1h,1,2"]
+|===
+| Type | `pointer/get` | Get an object model property value
+| Configuration
+| `string pointer` | The JSON pointer or JSON pointer template
+| Input value sockets
+| `int <segment>` | The JSON pointer template path segment to be substituted at runtime
```

Comment:
What does it mean to have the segment as an input integer value? Using the below example pointer `"/nodes/{myId}/scale"`, what does 0 mean? Is it referring to `/nodes`, the 0th item, and therefore 1 would refer to the `{myId}`? Or is it referring to the first instance of `{}`, so if there was `"/nodes/{myId}/{myProp}"` then 0 refers to `{myId}` and 1 would refer to `{myProp}`? But then what does the number mean, is it saying that only one of these will be substituted? Or is it saying to substitute `0` for `{myId}`? But then what's the text below about "the node has the `myId` input value socket"? If we had `"/nodes/{myId}/{myProp}"` then would the node need both `myId` and `myProp` inputs? If so, then we can perform the substitution with only the template and the inputs, and I don't know why we would ever read the `int <segment>` value. I genuinely don't understand what this means.

Reply:
I'm not the author but they way I read is the following:
if you have path template configured as `/nodes/{myId}/foo/{myProp}` then your node has two input sockets
`int myId` and `int myProp` respectively, so connecting node to values 42 and 100500 gives you `/nodes/42/foo/100500`. `int <segment>` is just not most readable way to convey idea of dynamic socket number, 
there is no 0s not 1s involved, curious how did you arrive at them?

Reply:
Oh! I completely missed that. So that is just a placeholder, and there is no input socket called segment. So for actual cases, there might be no segment input value sockets if it's not a template.